### PR TITLE
Add categorical generators

### PR DESCRIPTION
**Pull request by sarahmish**
1. add categorical generators and tests for LabelEncoder and Categorical transformers.
2. small tweak for LabelEncoder to allow input of type `np.ndarray`

### CODE REVIEW

File: tests/performance/datasets/categorical.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import numpy as np
+
+from tests.performance.datasets.base import BaseDatasetGenerator
+from tests.performance.datasets.utils import add_nans
+
+
+class RandomIntegerGenerator(BaseDatasetGenerator):
+    """Generator that creates an array of random integers."""
+
+    TYPE = 'numerical'
```
Comment (csala) [inline tests/performance/datasets/categorical.py:12]: Would this (and the type for the others) not be `categorical` instead?

File: tests/performance/datasets/categorical.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
+"""Dataset Generators for categorical transformers."""
```
Comment (csala) [inline tests/performance/datasets/categorical.py:1]: Should we add a dataset that produces a mixture of types?
We should also think about adding other types that can be used as categories, like `datetimes`, `floats` or `bools`.
Maybe it is not necessary to add explicitly datasets for each possible type, but at least have one dataset that produces a mixture of all of them, as well as NaNs and Nones.

**Review by csala — CHANGES_REQUESTED** [review]

**Review by csala — APPROVED** [review]
I think this looks great @sarahmish !
Any thoughts @amontanez24 ?

**Review by amontanez24 — APPROVED** [review]
I think this is good to merge!

**Pull request merged**